client/src/pages/StadiumProfile.tsx

```diff
@@ -1,10 +1,51 @@
 import { Box } from '@mui/system';
-import ImageSlider from '../components/stadiumProfile/ImageSlider';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import {
+  BioSection,
+  ImageSlider,
+  ProfileCalender,
+} from '../components/stadiumProfile';
+import { UserData } from '../interfaces/StadiumProfile';
 
 const StadiumProfile = () => {
+  const [userData, setUserData] = useState<UserData | null>(null);
+  const [gallery, setGallery] = useState({
+    id: 0,
+    image1: '',
+    image2: '',
+    image3: '',
+    image4: '',
+  });
+
+  const { id } = useParams();
+  const fetchProfileData = async (stadiumId: string) => {
```

Comment:
Wrap the  fetchProfileData function in a try-catch block to handle any errors that may occur during the API request and check for the length begore setting the Gallery
 const fetchProfileData = async (stadiumId: string) => {
    try {
      const response = await axios.get(`/api/v1/stadiums/profile/${stadiumId}`);
      const { data } = response.data;
      const { Stadium, ...restData } = data;

      if (Stadium && Stadium.stadiumGallery && Stadium.stadiumGallery.length > 0) {
        setGallery(Stadium.stadiumGallery[0]);
      }

      setUserData(restData as UserData);
    } catch (error) {
      // Handle error here
    }
  };